main/sensors/sensors.c

```diff
@@ -0,0 +1,107 @@
+#include "sensors.h"
+#include "adc.h"
+
+static uint64_t count = 0;
+static uint64_t prev_count = 0;
+static bool timer_started = false;
+static gptimer_handle_t timer_handle = NULL;
+static uint32_t ldr_data;
+static uint32_t prev_ldr_data;
+static uint32_t bat_data;
+
+static void interrupt_handler(void* args) {
+	if (!timer_started) {
+		gptimer_set_raw_count(timer_handle, 0);
```

Comment:
These aren't wrapped in PASS_ERROR, which makes sense, but they don't handle errors.
Please add an ELOG in case of an error, or write logic in case they do error